### PR TITLE
fix(TDI-38441): add failOn error to tFileCopy

### PR DESCRIPTION
**Pull request by dmytro-sylaiev**
* Create new checkbox property for tFileCopy component (default value - false)

* Create en message for it

* Change logic for component - it now trows RuntimeException if failOn is enabled

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 1]
```diff
@@ -152,21 +185,50 @@
...
+%>
+					org.talend.FileCopy.copyFile(srcFile_<%=cid %>.getPath(), desFile_<%=cid %>.getPath(), <%=reFile %>);
+<%
+			if (!failOn) {
+%>
+				} catch (Exception e) {
+<%
+					if (isLog4jEnabled) {
+%>
+						log.error("<%=cid %> " + e.getMessage());
```
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: Pass exception to log.error():
```java
log.error("<%=cid %> " + e.getMessage(), e);
```

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 2]
```diff
@@ -82,10 +84,21 @@
...
-									log.error("<%=cid%> - " + e.getMessage());
-									<%}%>
+<%
+									if (failOn) {
+%>
+										throw new RuntimeException("<%=cid%> - " + e.getMessage());
+<%
+									} else if(isLog4jEnabled) {
+%>
+										log.error("<%=cid%> - " + e.getMessage());
```
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: Pass exception to log.error():
```java
log.error("<%=cid %> " + e.getMessage(), e);
```

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 3]
```diff
@@ -100,11 +113,18 @@
 						desFolder.mkdirs();
 					} 
         		}else{
-        			System.err.println("Path name \""+rootFolder.getPath()+"\" does not denote a directory, or if an I/O error occurs.");
 <%
-					if(isLog4jEnabled) {
+					if (failOn) {
+%>
+						throw new RuntimeException("Path name \"" + rootFolder.getPath() + "\" does not denote a directory, or if an I/O error occurs.");
```
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: Code which constructs error message is duplicated, you can construct it once and then use in all cases.

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 4]
```diff
@@ -128,9 +148,22 @@
 
 		// here need check first, before mkdirs().
 		if (!srcFile_<%=cid %>.exists() || !srcFile_<%=cid %>.isFile()) {
-			throw new RuntimeException("The source File \"" + srcFileName_<%=cid %> + "\" does not exist or is not a file.");
+<%
+			if (failOn) {
+%>
+				throw new RuntimeException("The source File \"" + srcFileName_<%=cid %> + "\" does not exist or is not a file.");
```
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: Code which constructs error message is duplicated, you can construct it once and then use in all cases.

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 5]
```diff
@@ -152,21 +185,50 @@
...
 			if(reFile) {
 %>
 				java.io.File isRemoved_<%=cid %> = new java.io.File(<%=fileName%>);
 				if(isRemoved_<%=cid %>.exists()) {
 <%
-				if(isLog4jEnabled) {
+					if (failOn) {
 %>
-					log.error("<%=cid%> - The source file \"" + srcFileName_<%=cid %> + "\" could not be removed from the folder because it is open or you only have read-only rights.");
+						throw new RuntimeException("<%=cid%> - The source file \"" + srcFileName_<%=cid %> + "\" could not be removed from the folder because it is open or you only have read-only rights.");
```
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: Code which constructs error message is duplicated, you can construct it once and then use in all cases.

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet (hunk 3 above)
Comment (iistomin-gl) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: You could use String.format() with parameters to construct error message:
```java
String errorMessage = String.format(""<%=cid%> - Path name \"%s\" does not denote a directory, or if an I/O error occurs.", rootFolder.getPath());
...
throw new RuntimeException(errorMessage, e);
...
log.error(errorMessage, e);
...
System.err.println(errorMessage);
...
```

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet (hunk 2 above)
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: > Pass exception to log.error

Doesn't use in any component

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet (hunk 1 above)
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: > Pass exception to log.error

Doesn't use in any component

File: main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet [hunk 6]
```diff
@@ -152,21 +187,51 @@
 				}
 <%
 			}
+			if (!failOn) {
 %>           
-			org.talend.FileCopy.copyFile(srcFile_<%=cid %>.getPath(), desFile_<%=cid %>.getPath(), <%=reFile %>);
+				try {
```
Comment (lxia-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileCopy/tFileCopy_main.javajet]: You are adding try/catch here: behavior changed.

**Pull request merged**